src/io/github/sspanak/tt9/ime/KeyPadHandler.java

```diff
@@ -131,12 +130,13 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
 			event.startTracking();
 		}
 
-		return Key.isNumber(keyCode)
+		return
+			Key.isNumber(keyCode)
 			|| Key.isOK(keyCode)
-			|| Key.isHotkey(settings, keyCode) || Key.isHotkey(settings, -keyCode) // press or hold a hotkey
-			|| (keyCode == KeyEvent.KEYCODE_POUND && onText("#"))
-			|| (keyCode == KeyEvent.KEYCODE_STAR && onText("*"))
-			|| super.onKeyDown(keyCode, event); // let the system handle the keys we don't care about (usually, only: KEYCODE_BACK)
+			|| Key.isPoundOrStar(keyCode)
```

Comment:
Must not break typing `+` and `;` on Kyocera.
```suggestion
			|| (Key.isPoundOrStar(keyCode) && onText(String.valueOf( (char) event.getUnicodeChar(), validateOnly=true ));
```